datafusion/expr/src/udaf.rs

```diff
@@ -287,18 +286,23 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
     /// aggregate function was called.
     fn accumulator(&self, acc_args: AccumulatorArgs) -> Result<Box<dyn Accumulator>>;
 
+    /// Return the fields for the function
+    fn field(&self, _args: FieldArgs) -> Result<Field> {
```

Comment:
Is this the same information as `return_type`? If so, perhaps we should deprecate the `return_type` function?

Reply:
`return_type` is part of the information in `field` and `state_field` not replaceable to `field `